zarr/storage.py

```diff
@@ -1414,9 +1414,15 @@ def getitems(
         self, keys: Sequence[str], *, contexts: Mapping[str, Context]
     ) -> Mapping[str, Any]:
         keys_transformed = [self._normalize_key(key) for key in keys]
-        results = self.map.getitems(keys_transformed, on_error="omit")
-        # The function calling this method may not recognize the transformed keys
-        # So we send the values returned by self.map.getitems back into the original key space.
+        results = self.map.getitems(keys_transformed, on_error="return")
+        # Only recognized errors will prompt KeyErrors in the function calling this method
+        for k, v in results.copy().items():
```

Comment:
It would be nice to rework this to avoid the `copy` here.

Reply:
With needing to return a new dictionary (with untransformed keys), there was essentially already a shallow copy. Re-organized to take advantage of that and got good speedup.

Reply:
In my view this is resolved, but I'm leaving the comment open for further consideration or approval.

Reply:
Right, this is not copying the data blobs, so I don't think it's a problem.